app/src/app/store/mapStore.ts

```diff
@@ -69,6 +78,7 @@ export const useMapStore = create<MapStore>((set) => ({
     set((state) => {
       if (!state.mapRef) return { selectedLayer: null };
       addBlockLayers(state.mapRef, layer);
```

Comment:
I'm not sure this is the issue exactly but I worry that there are collisions with the map state if changes are also coming from the zustand store. We should be using `subscribe`s for this: https://github.com/pmndrs/zustand?tab=readme-ov-file#using-subscribe-with-selector

Reply:
Makes sense and this is much nicer.